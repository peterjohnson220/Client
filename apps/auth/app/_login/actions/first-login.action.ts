import { Action } from '@ngrx/store';

export const VALIDATE_FIRST_LOGIN = '[Auth/First Login] Validate First Login';
export const VALIDATE_FIRST_LOGIN_SUCCESS = '[Auth/First Login] Validate First Login Success';
export const VALIDATE_FIRST_LOGIN_ERROR = '[Auth/First Login] Validate First Login Error';
export const FIRST_LOGIN_UPDATING_PASSWORD = '[Auth/First Login] Updating Password';
export const FIRST_LOGIN_UPDATING_PASSWORD_SUCCESS = '[Auth/First Login] Updating Password Success';
export const FIRST_LOGIN_UPDATING_PASSWORD_ERROR = '[Auth/First Login] Updating Password Error';

export class ValidateFirstLogin implements Action {
  readonly type = VALIDATE_FIRST_LOGIN;
  constructor() {}
}

export class ValidateFirstLoginSuccess implements Action {
  readonly type = VALIDATE_FIRST_LOGIN_SUCCESS;
  constructor() {}
}

export class ValidateFirstLoginError implements Action {
  readonly type = VALIDATE_FIRST_LOGIN_ERROR;
  constructor() {}
}

export class UpdatePassword implements Action {
  readonly type = FIRST_LOGIN_UPDATING_PASSWORD;
  constructor(public payload: any) {}
}

export class UpdatePasswordSuccess implements Action {
  readonly type = FIRST_LOGIN_UPDATING_PASSWORD_SUCCESS;
  constructor() {}
}

export class UpdatePasswordError implements Action {
  readonly type = FIRST_LOGIN_UPDATING_PASSWORD_ERROR;
  constructor(public payload: any) {}
}

export type Actions
  = ValidateFirstLogin
  | ValidateFirstLoginSuccess
  | ValidateFirstLoginError
  | UpdatePassword
  | UpdatePasswordSuccess
  | UpdatePasswordError;
