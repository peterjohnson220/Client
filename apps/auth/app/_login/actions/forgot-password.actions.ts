import { Action } from '@ngrx/store';

export const SENDING_PASSWORD_RESET  = '[auth] Sending Password Reset';
export const SENDING_PASSWORD_RESET_SUCCESS  = '[auth] Sending Password Reset Success';
export const SENDING_PASSWORD_RESET_ERROR  = '[auth] Sending Password Reset Error';

export class SendingPasswordReset implements Action {
  readonly type = SENDING_PASSWORD_RESET;

  constructor(public payload: any) {}
}

export class SendingPasswordResetSuccess implements Action {
  readonly type = SENDING_PASSWORD_RESET_SUCCESS;

  constructor() {}
}

export class SendingPasswordResetError implements Action {
  readonly type = SENDING_PASSWORD_RESET_ERROR;
  readonly payload;
}

export type Actions
  = SendingPasswordReset
  | SendingPasswordResetSuccess
  | SendingPasswordResetError;
