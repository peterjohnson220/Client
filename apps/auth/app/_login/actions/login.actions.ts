import { Action } from '@ngrx/store';

export const LOGIN = '[Auth/Login] Login';
export const LOGIN_SUCCESS = '[Auth/Login] Validate Login Success';
export const LOGIN_ERROR = '[Auth/Login] Validate Login Error';
export const LOGIN_401_ERROR = '[Auth/Login] Validate Login 401 Error';

export class Login implements Action {
  readonly type = LOGIN;
  constructor(public payload: any) {}
}

export class LoginSuccess implements Action {
  readonly type = LOGIN_SUCCESS;
  constructor() {}
}

export class LoginError implements Action {
  readonly type = LOGIN_ERROR;
  constructor(public error: any) {}
}

export class Login401Error implements Action {
  readonly type = LOGIN_401_ERROR;
  constructor() { }
}
export type Actions
  = Login
  | LoginSuccess
  | LoginError
  | Login401Error;
