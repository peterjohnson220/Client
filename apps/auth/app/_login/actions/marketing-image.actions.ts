import { Action } from '@ngrx/store';
import { MarketingImageDto } from 'libs/models/marketing/marketing-image-dto.model';

export const GET_MARKETING_IMAGE =  '[marketing/marketing-image] Get marketing image';
export const GET_MARKETING_IMAGE_SUCCESS =  '[marketing/marketing-image] Get marketing image success';
export const GET_MARKETING_IMAGE_ERROR =  '[marketing/marketing-image] Get marketing image error';

export class GetMarketingImage implements Action {
    readonly type = GET_MARKETING_IMAGE;
  }

export class GetMarketingImageSuccess implements Action {
    readonly type = GET_MARKETING_IMAGE_SUCCESS;
    constructor(public payload: MarketingImageDto) {
    }
  }

export class GetMarketingImageError implements Action {
    readonly type = GET_MARKETING_IMAGE_ERROR;
    constructor(public payload: Error) {}
  }

  export type Actions
  = GetMarketingImage
  | GetMarketingImageSuccess
  | GetMarketingImageError;

