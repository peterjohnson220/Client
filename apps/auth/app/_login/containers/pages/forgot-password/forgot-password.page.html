<div class="content p-4" *ngIf="!formSubmitSuccess && !submissionError">
  <h1 class="mb-5">Reset Password</h1>
  <p>Please enter your email address and we will send you a password reset link.</p>
  <form [formGroup]="emailForm" #formDir="ngForm">
    <div class="reset-box ml-auto mr-auto mt-4 mb-3">
      <label id="lblEmail" class="label label-default float-left font-weight-bold" for="txtEmail">EMAIL ADDRESS</label>
      <input id="txtEmail" type="email" class="form-control" placeholder="johndoe@notmail.com" name="email"
             formControlName="email">
      <div *ngIf="attemptedFormSubmit && email.invalid"
           class="alert alert-danger">
        <div *ngIf="email.errors.required">
          Email address is required.
        </div>
        <div *ngIf="email.errors.pattern">
          Please enter a valid email address.
        </div>
      </div>
    </div>
    <button type="button" id="Reset" class="btn btn-primary"
            [disabled]="(emailForm.invalid && attemptedFormSubmit) || formSubmitting || !settingsSuccess"
            [class.disabled]="(emailForm.invalid && attemptedFormSubmit) || formSubmitting" (click)="submit()">
      <span *ngIf="!formSubmitting">RESET PASSWORD</span>
      <span *ngIf="formSubmitting">PROCESSING...</span>
    </button>
  </form>
</div>
<div class="content p-4" *ngIf="formSubmitSuccess">
  <h1>Password Reset Sent!</h1>
  <p class="mt-4">We've sent you an email containing a link that will allow you to reset your password for the next 24
    hours. <br/>
    Please check your spam folder if the email doesn't appear within a few minutes.</p>
  <a href="/" id="ReturnToLogin" class="btn btn-primary mt-3">
    Return to Login
  </a>
</div>
<div class="content p-4" *ngIf="submissionError">
  <p class="text-danger">Sorry we couldn't process this request, please contact your system administrator.</p>
</div>
