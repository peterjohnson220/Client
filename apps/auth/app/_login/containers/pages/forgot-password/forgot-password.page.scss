.content {
  text-align: center;
  height: 300px;
}

h1 {
  color: #FFB300;
}

.reset-box {
  width: 300px;
}

.btn {
  background-color: #0883BE;
  border: 0;
}

.disabled {
  cursor: not-allowed;
}
