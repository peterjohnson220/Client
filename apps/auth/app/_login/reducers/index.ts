import { createSelector, createFeatureSelector } from '@ngrx/store';

// Import root app reducer
import * as fromRoot from 'libs/state/state';

// Import feature reducers
import * as fromForgotPasswordReducer from './forgot-password.reducer';
import * as fromFirstLoginReducer from './first-login-reducer';
import * as fromResetPasswordReducer from './reset-password.reducer';
import * as fromMarketingReducer from './marketing-image.reducer';
import * as fromLoginReducer from './login.reducer';
import * as fromSelfRegistrationReducer from './self-registration.reducer';

// Feature area state
export interface AuthMainState {
  forgotPassword: fromForgotPasswordReducer.State;
  resetPassword: fromResetPasswordReducer.State;
  firstLogin: fromFirstLoginReducer.State;
  marketingImage: fromMarketingReducer.State;
  login: fromLoginReducer.State;
  selfRegistration: fromSelfRegistrationReducer.State;
}

// Extend root state with feature area state
export interface State extends fromRoot.State {
  authMain: AuthMainState;
}

// Feature area reducers
export const reducers = {
  forgotPassword: fromForgotPasswordReducer.reducer,
  resetPassword: fromResetPasswordReducer.reducer,
  firstLogin: fromFirstLoginReducer.reducer,
  marketingImage: fromMarketingReducer.reducer,
  login: fromLoginReducer.reducer,
  selfRegistration: fromSelfRegistrationReducer.reducer
};

// Select Feature Area
export const selectAuthMainState = createFeatureSelector<AuthMainState>('authMain');

// Forgot password
export const forgotPasswordState =
  createSelector(selectAuthMainState, (state: AuthMainState) => state.forgotPassword);

export const getForgotPasswordSending =
  createSelector(forgotPasswordState, fromForgotPasswordReducer.getSendingPassword);
export const getForgotPasswordSuccess =
  createSelector(forgotPasswordState, fromForgotPasswordReducer.getSendingPasswordSuccess);
export const getForgotPasswordError =
  createSelector(forgotPasswordState, fromForgotPasswordReducer.getSendingPasswordResetError);

// First Login
export const firstLoginState =
  createSelector(selectAuthMainState, (state: AuthMainState) => state.firstLogin);

export const getValidatingFirstLogin =
  createSelector(firstLoginState, fromFirstLoginReducer.getValidatingFirstLogin);
export const getValidatingFirstLoginSuccess =
  createSelector(firstLoginState, fromFirstLoginReducer.getValidatingFirstLoginSuccess);
export const getValidatingFirstLoginError =
  createSelector(firstLoginState, fromFirstLoginReducer.getValidatingFirstLoginError);
export const getFirstLoginUpdatingPassword =
  createSelector(firstLoginState, fromFirstLoginReducer.getUpdatingPassword);
export const getFirstLoginUpdatingPasswordSuccess =
  createSelector(firstLoginState, fromFirstLoginReducer.getUpdatingPasswordSuccess);
export const getFirstLoginUpdatingPasswordError =
  createSelector(firstLoginState, fromFirstLoginReducer.getUpdatingPasswordError);

// Reset Password
export const resetPasswordState =
  createSelector(selectAuthMainState, (state: AuthMainState) => state.resetPassword);

export const getResettingPassword =
  createSelector(resetPasswordState, fromResetPasswordReducer.getResettingPassword);
export const getResetPasswordSuccess =
  createSelector(resetPasswordState, fromResetPasswordReducer.getResettingPasswordSuccess);
export const getResetPasswordTokenExpired =
  createSelector(resetPasswordState, fromResetPasswordReducer.getResettingPasswordTokenExpired);
export const getResetPasswordError =
  createSelector(resetPasswordState, fromResetPasswordReducer.getResettingPasswordError);
export const getCheckingResetPasswordToken =
  createSelector(resetPasswordState, fromResetPasswordReducer.getCheckingResetPasswordToken);
export const getCheckingResetPasswordTokenSuccess =
  createSelector(resetPasswordState, fromResetPasswordReducer.getCheckingResetPasswordTokenSuccess);
export const getPasswordMinimumLength =
  createSelector(resetPasswordState, fromResetPasswordReducer.getPasswordMinimumLength);


// marketing
export const getMarketingState =
  createSelector(selectAuthMainState, (state: AuthMainState) => state.marketingImage);

export const getMarketingImage =
  createSelector(getMarketingState, fromMarketingReducer.getMarketingImage);
export const getGettingMarketingImage =
  createSelector(getMarketingState, fromMarketingReducer.getGettingMarketingImage);
export const getGettingMarketingImageError =
  createSelector(getMarketingState, fromMarketingReducer.getGettingMarketingImageError);
export const getGettingMarketingImageSuccess =
  createSelector(getMarketingState, fromMarketingReducer.getGettingMarketingImageSuccess);

// login
export const loginState =
  createSelector(selectAuthMainState, (state: AuthMainState) => state.login);

export const getLogin =
  createSelector(loginState, fromLoginReducer.getLogin);
export const getLoginSuccess =
  createSelector(loginState, fromLoginReducer.getLoginSuccess);
export const getLoginError =
  createSelector(loginState, fromLoginReducer.getLoginError);
export const getShowSelfRegistrationForm =
  createSelector(loginState, fromLoginReducer.getShowSelfRegistrationForm);

// request access/self reg
export const selfRegistrationState =
  createSelector(selectAuthMainState, (state: AuthMainState) => state.selfRegistration);
export const getSelfRegistrationForm =
  createSelector(selfRegistrationState, fromSelfRegistrationReducer.getSelfRegistrationForm);
export const getSelfRegistrationIsSubmitting =
  createSelector(selfRegistrationState, fromSelfRegistrationReducer.getIsSubmitting);
export const getSelfRegistrationSubmitSuccess =
  createSelector(selfRegistrationState, fromSelfRegistrationReducer.getSubmitSuccess);
export const getSelfRegistrationSubmitError =
  createSelector(selfRegistrationState, fromSelfRegistrationReducer.getSubmitError);
