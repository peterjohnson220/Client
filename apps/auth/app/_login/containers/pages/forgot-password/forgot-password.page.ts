import { Component, OnInit } from '@angular/core';
import { FormControl, FormGroup, Validators} from '@angular/forms';

import { Store } from '@ngrx/store';
import { Observable, Subscription } from 'rxjs';

import * as fromForgotPasswordReducer from '../../../reducers';
import * as fromForgotPasswordActions from '../../../actions/forgot-password.actions';

@Component({
  selector: 'pf-forgot-password-page',
  templateUrl: './forgot-password.page.html',
  styleUrls: [ './forgot-password.page.scss' ]
})
export class ForgotPasswordPageComponent implements OnInit {

  loadingSubscription: Subscription;
  errorSubscription: Subscription;
  successSubscription: Subscription;

  submitting$: Observable<boolean>;
  submitError$: Observable<boolean>;
  submitSuccess$: Observable<boolean>;

  submissionError = false;
  formSubmitting = false;

  formControlEmail = '';
  attemptedFormSubmit = false;
  formSubmitSuccess = false;
  emailForm: FormGroup;
  emailRegex = /^\w+([\-\.+]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/;

  constructor(public store: Store<fromForgotPasswordReducer.State>) {

    this.submitting$ = this.store.select(fromForgotPasswordReducer.getForgotPasswordSending);
    this.submitError$ = this.store.select(fromForgotPasswordReducer.getForgotPasswordError);
    this.submitSuccess$ = this.store.select(fromForgotPasswordReducer.getForgotPasswordSuccess);
  }

  ngOnInit(): void {
    this.emailForm = new FormGroup({
      'email': new FormControl(this.formControlEmail, [
        Validators.required,
        Validators.pattern(this.emailRegex)
      ])
    });

    this.loadingSubscription = this.submitting$.subscribe(value => {
      if (value) {
        this.formSubmitting = true;
      }
    });

    this.errorSubscription = this.submitError$.subscribe(error => {
      if (error) {
        this.submissionError = true;
      }
    });

    this.successSubscription = this.submitSuccess$.subscribe(value => {
      if (value) {
        this.formSubmitSuccess = true;
      }
    });
  }

  submit() {

    this.attemptedFormSubmit = true;
    if (!this.emailForm.invalid) {
       this.store.dispatch(new fromForgotPasswordActions.SendingPasswordReset({ email: this.email.value.toString() }));
    }
  }

  get email() {
    return this.emailForm.get('email');
  }
}
