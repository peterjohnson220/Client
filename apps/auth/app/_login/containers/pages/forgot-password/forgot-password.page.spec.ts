import { NO_ERRORS_SCHEMA } from '@angular/core';
import { TestBed, ComponentFixture } from '@angular/core/testing';

import { StoreModule, Store, combineReducers } from '@ngrx/store';
import { of } from 'rxjs';
import spyOn = jest.spyOn;

import { AbstractFeatureFlagService } from 'libs/core/services/feature-flags';
import * as fromRootState from 'libs/state/state';

import * as fromForgotPasswordActions from '../../../actions/forgot-password.actions';
import { ForgotPasswordPageComponent } from '../forgot-password/forgot-password.page';
import * as fromReducers from '../../../reducers';

import {
  ReactiveFormsModule,
  FormBuilder
} from '@angular/forms';

describe('Auth - Forgot Password', () => {
  let fixture: ComponentFixture<ForgotPasswordPageComponent>;
  let instance: ForgotPasswordPageComponent;
  let store: Store<fromReducers.State>;
  let formBuilder: FormBuilder;

  // Configure Testing Module for before each test
  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [
        StoreModule.forRoot({
          ...fromRootState.reducers,
          forgotPasswordReducers: combineReducers(fromReducers.reducers)
        }),
        ReactiveFormsModule
      ],
      declarations: [
        ForgotPasswordPageComponent
      ],
      providers: [
        {
          provide: AbstractFeatureFlagService,
          useValue: { enabled: jest.fn(), bindEnabled: jest.fn() }
        }
      ],
      // Shallow Testing
      schemas: [ NO_ERRORS_SCHEMA ]
    });
    formBuilder = new FormBuilder();
    fixture = TestBed.createComponent(ForgotPasswordPageComponent);
    instance = fixture.componentInstance;
    instance.emailForm = formBuilder.group({
      'email': ['test@test.com']
    });

    store = TestBed.inject(Store);
    spyOn(store, 'dispatch');
  });

  it('should show reset password section when page is loaded', () => {

   instance.submitting$ = of(false);
   instance.submitSuccess$ = of(false);
   instance.submitError$ = of(false);

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should disable reset password button and text when submitting', () => {

    instance.submitting$ = of(true);
    instance.submitSuccess$ = of(false);
    instance.submitError$ = of(false);

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should show email sent notification on submitSuccess', () => {

    instance.submitting$ = of(false);
    instance.submitSuccess$ = of(true);
    instance.submitError$ = of(false);

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });


  it('should show error notification on submitError', () => {

    instance.submitting$ = of(false);
    instance.submitSuccess$ = of(false);
    instance.submitError$ = of(true);

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });


  it('should dispatch SendingPasswordReset when reset password button is clicked', () => {

    const testEmail = 'test@test.com';

    instance.submitting$ = of(false);
    instance.submitSuccess$ = of(false);
    instance.submitError$ = of(false);

    const clientSettingRequest = {
      email: testEmail,
      clientCaptchaToken: '',
      clientCaptchaSiteKey: undefined
    };

    const action = new fromForgotPasswordActions.SendingPasswordReset(clientSettingRequest);

    instance.submit();

    expect(store.dispatch).toHaveBeenCalledWith(action);
  });

});
