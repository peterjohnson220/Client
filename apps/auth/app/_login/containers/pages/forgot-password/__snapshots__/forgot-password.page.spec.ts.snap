// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth - Forgot Password should disable reset password button and text when submitting 1`] = `
<pf-forgot-password-page
  attemptedFormSubmit="false"
  emailForm={[Function FormGroup]}
  emailRegex={[Function RegExp]}
  errorSubscription={[Function Subscriber]}
  formControlEmail=""
  formSubmitSuccess="false"
  formSubmitting={[Function Boolean]}
  loadingSubscription={[Function Subscriber]}
  store={[Function Store]}
  submissionError="false"
  submitError$={[Function Observable]}
  submitSuccess$={[Function Observable]}
  submitting$={[Function Observable]}
  successSubscription={[Function Subscriber]}
>
  <div
    class="content p-4"
  >
    <h1
      class="mb-5"
    >
      Reset Password
    </h1>
    <p>
      Please enter your email address and we will send you a password reset link.
    </p>
    <form
      class="ng-untouched ng-pristine ng-invalid"
      ng-reflect-form="[object Object]"
      novalidate=""
    >
      <div
        class="reset-box ml-auto mr-auto mt-4 mb-3"
      >
        <label
          class="label label-default float-left font-weight-bold"
          for="txtEmail"
          id="lblEmail"
        >
          EMAIL ADDRESS
        </label>
        <input
          class="form-control ng-untouched ng-pristine ng-invalid"
          formcontrolname="email"
          id="txtEmail"
          name="email"
          ng-reflect-name="email"
          placeholder="johndoe@notmail.com"
          type="email"
        />
        
      </div>
      <button
        class="btn btn-primary disabled"
        disabled=""
        type="button"
      >
        
        
        <span>
          PROCESSING...
        </span>
      </button>
    </form>
  </div>
</pf-forgot-password-page>
`;

exports[`Auth - Forgot Password should show email sent notification on submitSuccess 1`] = `
<pf-forgot-password-page
  attemptedFormSubmit="false"
  emailForm={[Function FormGroup]}
  emailRegex={[Function RegExp]}
  errorSubscription={[Function Subscriber]}
  formControlEmail=""
  formSubmitSuccess={[Function Boolean]}
  formSubmitting="false"
  loadingSubscription={[Function Subscriber]}
  store={[Function Store]}
  submissionError="false"
  submitError$={[Function Observable]}
  submitSuccess$={[Function Observable]}
  submitting$={[Function Observable]}
  successSubscription={[Function Subscriber]}
>
  <div
    class="content p-4"
  >
    <h1>
      Password Reset Sent!
    </h1>
    <p
      class="mt-4"
    >
      We've sent you an email containing a link that will allow you to reset your password for the next 24 hours. 
      <br />
       Please check your spam folder if the email doesn't appear within a few minutes.
    </p>
    <a
      class="btn btn-primary mt-3"
      href="/"
    >
       Return to Login 
    </a>
  </div>
</pf-forgot-password-page>
`;

exports[`Auth - Forgot Password should show error notification on submitError 1`] = `
<pf-forgot-password-page
  attemptedFormSubmit="false"
  emailForm={[Function FormGroup]}
  emailRegex={[Function RegExp]}
  errorSubscription={[Function Subscriber]}
  formControlEmail=""
  formSubmitSuccess="false"
  formSubmitting="false"
  loadingSubscription={[Function Subscriber]}
  store={[Function Store]}
  submissionError={[Function Boolean]}
  submitError$={[Function Observable]}
  submitSuccess$={[Function Observable]}
  submitting$={[Function Observable]}
  successSubscription={[Function Subscriber]}
>
  <div
    class="content p-4"
  >
    <p
      class="text-danger"
    >
      Sorry we couldn't process this request, please contact your system administrator.
    </p>
  </div>
</pf-forgot-password-page>
`;

exports[`Auth - Forgot Password should show reset password section when page is loaded 1`] = `
<pf-forgot-password-page
  attemptedFormSubmit="false"
  emailForm={[Function FormGroup]}
  emailRegex={[Function RegExp]}
  errorSubscription={[Function Subscriber]}
  formControlEmail=""
  formSubmitSuccess="false"
  formSubmitting="false"
  loadingSubscription={[Function Subscriber]}
  store={[Function Store]}
  submissionError="false"
  submitError$={[Function Observable]}
  submitSuccess$={[Function Observable]}
  submitting$={[Function Observable]}
  successSubscription={[Function Subscriber]}
>
  <div
    class="content p-4"
  >
    <h1
      class="mb-5"
    >
      Reset Password
    </h1>
    <p>
      Please enter your email address and we will send you a password reset link.
    </p>
    <form
      class="ng-untouched ng-pristine ng-invalid"
      ng-reflect-form="[object Object]"
      novalidate=""
    >
      <div
        class="reset-box ml-auto mr-auto mt-4 mb-3"
      >
        <label
          class="label label-default float-left font-weight-bold"
          for="txtEmail"
          id="lblEmail"
        >
          EMAIL ADDRESS
        </label>
        <input
          class="form-control ng-untouched ng-pristine ng-invalid"
          formcontrolname="email"
          id="txtEmail"
          name="email"
          ng-reflect-name="email"
          placeholder="johndoe@notmail.com"
          type="email"
        />
        
      </div>
      <button
        class="btn btn-primary"
        type="button"
      >
        
        <span>
          RESET PASSWORD
        </span>
        
      </button>
    </form>
  </div>
</pf-forgot-password-page>
`;
